### PR TITLE
Add check for undefined window and navigator as well as tests

### PR DESCRIPTION
**Pull request by myty**
Fixes issue #123 

- [x] Related GitHub issue(s) linked in PR description
- [x] Destination branch merged, built and tested with your changes
- [x] Code formatted and follows best practices and patterns
- [x] Code builds cleanly (no _additional_ warnings or errors)
- [x] Manually tested
- [x] Automated tests are passing
- [x] No _decreases_ in automated test coverage
- N/A Documentation updated (readme, docs, comments, etc.)
- N/A Localization: No hard-coded error messages in code files (_minimally_ in string constants)

### CODE REVIEW

File: src/utilities/network-information-utils.ts [hunk 1]
```diff
@@ -5,24 +5,32 @@ import { Navigator } from "../types/navigator";
...
                 undefined
             );
         },
 
         /**
          * Returns a `NetworkConnection` object which is an aggregate of `navigator.connection` and `navigator.onLine`
          */
-        getNetworkConnection() {
+        getNetworkConnection():
+            | (NetworkInformation & { isOnline: boolean })
```
Comment (brandongregoryscott) [inline src/utilities/network-information-utils.ts:28]: Is there an interface that represents this? If so, I'd swap it instead of redefining here

**Review by brandongregoryscott — APPROVED** [review]
Just one comment I'd like some clarification on. Other than that, good to go!

File: src/utilities/network-information-utils.ts (hunk 1 above)
Comment (myty) [reply src/utilities/network-information-utils.ts:28]: Yeah.  I don't think there's a type right now, but probably a good idea to add one in.

Comment (brandongregoryscott) [reply src/utilities/network-information-utils.ts:28]: That was my guess based on the JS Doc - that should be updated too.

**Review by brandongregoryscott — APPROVED** [review]

**Review by jhugs — APPROVED** [review]

**Pull request merged**